lib/backends/cpu/ops/upsample.ts

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import {Upsample} from '../../../ops/upsample';
+import {Tensor} from '../../../tensor';
+import {CpuInferenceHandler} from '../inference-handler';
+
+export class CpuUpsample extends Upsample {
+  run(inferenceHandler: CpuInferenceHandler, inputs: Tensor[]): Tensor[] {
+    const xDims = inputs[0].dims;
+    const yDims = xDims.map((dim, i) => Math.floor(dim * this.scales[i]));
+    const y = new Tensor(yDims, inputs[0].type);
+    if (this.mode === 'nearest') {
+      upsampleNearest(inputs[0].data, y.data, xDims, yDims, this.scales);
+    } else {
+      upsampleLinear(inputs[0].data, y.data, xDims, yDims, this.scales, this.roi);
```

Comment:
Seems like roi is not used. What is this for?

Reply:
Explanation of why -
 - There are different opset (7, 9, 10) of `Upsample` in ONNX spec, in this change we only implement [7-8]. Our goal is to support all opset version but we have priorities so this time the change is for [7-8] only.
- in opset v10, `Upsample` is deprecated. It's replaced by `Resize`, which is designed to support both upsample and downsample. It also introduced more attributes, in which `roi` is one of them.
- definition of Upsample does not have a roi input, however, Resize does. the cpu backend implementation is basically referencing implementation of onnxruntime, which uses a general implementation that works for Upsample's each opset version, and Resize as well.

I think I can remove the roi here for now and add it back when implementing other opsets.